include/mata/nfa.hh

```diff
@@ -806,6 +718,64 @@ public:
     void add_symbols_to(OnTheFlyAlphabet& alphabet);
 }; // Nfa
 
+/**
+  * Fill @p alphabet with symbols from @p nfa.
+  * @param[in] nfa NFA with symbols to fill @p alphabet with.
+  * @param[out] alphabet Alphabet to be filled with symbols from @p nfa.
+  */
+void fill_alphabet(const Mata::Nfa::Nfa& nfa, Mata::OnTheFlyAlphabet& alphabet);
+
+// Adapted from: https://www.fluentcpp.com/2019/01/25/variadic-number-function-parameters-type/.
+template<bool...> struct bool_pack{};
+/// Checks for all types in the pack.
+template<typename... Ts> using conjunction = std::is_same<bool_pack<true,Ts::value...>, bool_pack<Ts::value..., true>>;
+/// Checks whether all types are 'Nfa'.
+template<typename... Ts> using AreAllNfas = typename conjunction<std::is_same<Ts, const Mata::Nfa::Nfa&>...>::type;
+
+/**
+ * Create alphabet from variable number of NFAs.
+ * @tparam[in] Nfas Type Nfa.
+ * @param[in] nfas NFAs to create alphabet from.
+ * @return Created alphabet.
+ */
+template<typename... Nfas, typename = AreAllNfas<Nfas...>>
```

Comment:
I haven't noticed it before but what does the typename AreAllNfas? Is it used somewhere?

Reply:
This is a typename construct which at compile time checks, that all arguments passed to this function are of type `Nfa`. The whole section with bool packs and definition of `AreAllNfas` is there only so that one cannot call this function with invalid parameters such as `create_alphabet(1, "foo", Afa{}, ...)`. Without this typename, one could pass anything to the function, which would promptly fail when executing the lambda function within.